### PR TITLE
Docs step/stepi/next

### PR DESCRIPTION
**Pull request by 1e-to**
Documentation about commands step/stepi/next

### CODE REVIEW

**Comment by diptorupd** [comment]
@1e-to please provide a description for the PR.

File: docs/user_guides/debugging/stepping.rst [hunk 1]
```diff
@@ -28,70 +23,489 @@ Consider ``numba-dppy`` kernel code:
...
-    [Detaching after fork from child process 10505]
-    [New Thread 0x7fffd601f700 (LWP 10513)]
-    intelgt: gdbserver-gt started for process 10491.
-    intelgt: attached to device 1 of 1; id 0x5927 (Gen9)
-    [New Thread 0x7fffc593d700 (LWP 10530)]
-    [Detaching after fork from child process 10531]
-    compile_kernel (array(float32, 1d, C), array(float32, 1d, C), array(float32, 1d, C))
-    [Detaching after fork from child process 10532]
-    [Detaching after fork from child process 10533]
+    [Detaching after fork from child process 1691392]
```
Comment (PokhodenkoSA) [inline docs/user_guides/debugging/stepping.rst:95]: Too much text. Keep only important parts.

File: docs/user_guides/debugging/stepping.rst [hunk 2]
```diff
@@ -28,70 +23,489 @@ Consider ``numba-dppy`` kernel code:
...
-    Thread 2.3 hit Breakpoint 1, with SIMD lanes [0-1], dppy_py_devfn__5F__5F_main_5F__5F__2E_data_5F_parallel_5F_sum_24_1_2E_array_28_float32_2C__20_1d_2C__20_C_29__2E_array_28_float32_2C__20_1d_2C__20_C_29__2E_array_28_float32_2C__20_1d_2C__20_C_29_ () at stepping.py:7
+    Thread 2.3 hit Breakpoint 1, with SIMD lanes [0-1], __main__::data_parallel_sum () at sum.py:7
     7           i = dppy.get_global_id (0)  # numba-kernel-breakpoint
     (gdb) s
     8           l1 = a[i]                   # second-line
     (gdb) s
     9           l2 = b[i]                   # third-line
     (gdb) s
     10          c[i] = l1 + l2              # fourth-line
+    (gdb) c
```
Comment (PokhodenkoSA) [inline docs/user_guides/debugging/stepping.rst:143]: Use full command (i.e. continue).

**Review by PokhodenkoSA — CHANGES_REQUESTED** [review]
Very much unnecessary details in logs. Command should not be shortened (i.e. c -> continue, b -> break

File: docs/user_guides/debugging/stepping.rst [hunk 3]
```diff
@@ -40,58 +22,122 @@ Run debugger:
...
 
-    [Switching to Thread 1.1073742080 lane 0]
+    (gdb) break simple_sum.py:22
+    Breakpoint 1 (simple_sum.py:22) pending.
+    (gdb) run
+    Thread 2.2 hit Breakpoint 1, with SIMD lanes [0-7], __main__::data_parallel_sum () at simple_sum.py:22
+    22          i = dppy.get_global_id(0)
+    (gdb) step
+    Thread 2.3 hit Breakpoint 1, with SIMD lanes [0-1], __main__::data_parallel_sum () at simple_sum.py:22
+    22          i = dppy.get_global_id(0)
```
Comment (PokhodenkoSA) [inline docs/user_guides/debugging/stepping.rst:38]: This is wrong behavior we waiting to be fixed in numba.

File: docs/user_guides/debugging/stepping.rst [hunk 4]
```diff
@@ -40,58 +22,122 @@ Run debugger:
...
+.. code-block:: bash
+
+    (gdb) break simple_dppy_func.py:28
+    Breakpoint 1 (simple_dppy_func.py:28) pending.
+    (gdb) run
+    Thread 2.2 hit Breakpoint 1, with SIMD lanes [0-7], __main__::kernel_sum () at simple_dppy_func.py:28
+    28          i = dppy.get_global_id(0)
+    (gdb) step
+    Thread 2.3 hit Breakpoint 1, with SIMD lanes [0-1], __main__::kernel_sum () at simple_dppy_func.py:28
+    28          i = dppy.get_global_id(0)
```
Comment (PokhodenkoSA) [inline docs/user_guides/debugging/stepping.rst:64]: Also wrong behavior.

File: docs/user_guides/debugging/stepping.rst (hunk 3 above)
Comment (PokhodenkoSA) [reply docs/user_guides/debugging/stepping.rst:38]: But possibly it is related to breakpoint mode on GPU. We need to have an answer why it is necessary,

File: docs/user_guides/debugging/stepping.rst [hunk 5]
```diff
@@ -40,58 +22,122 @@ Run debugger:
...
+    gdb-oneapi -q --args python simple_dppy_func.py
+
+GDB output:
+
+.. code-block:: bash
+
+    (gdb) break simple_dppy_func.py:28
+    Breakpoint 1 (simple_dppy_func.py:28) pending.
+    (gdb) run
+    Thread 2.2 hit Breakpoint 1, with SIMD lanes [0-7], __main__::kernel_sum () at simple_dppy_func.py:28
```
Comment (PokhodenkoSA) [inline docs/user_guides/debugging/stepping.rst:135]: Use `...` to indicate that you skipped some part of output.

**Review by PokhodenkoSA — CHANGES_REQUESTED** [review]

File: docs/user_guides/debugging/stepping.rst [hunk 6]
```diff
@@ -40,58 +27,162 @@ Run debugger:
...
+
+.. _single_stepping:
+
+``set scheduler-locking step``
+-------------------------------
+
+Debug of the first line of the kernel and functions works out twice.
+This happens because you are debugging a multi-threaded program and multiple events may be received from different threads.
+This is the default behavior, but you can configure it for more efficient debugging.
+To ensure the current thread executes a single line without interference, set the scheduler-locking setting to on or step:
```
Comment (PokhodenkoSA) [inline docs/user_guides/debugging/stepping.rst:157]: Is it from other documentation? Could you provide a link?

File: docs/user_guides/debugging/stepping.rst [hunk 7]
```diff
@@ -40,58 +27,162 @@ Run debugger:
...
+    29          c_in_kernel[i] = func_sum(a_in_kernel[i], b_in_kernel[i])
+    (gdb) next
+    Done...
+
+.. _single_stepping:
+
+``set scheduler-locking step``
+-------------------------------
+
+Debug of the first line of the kernel and functions works out twice.
```
Comment (PokhodenkoSA) [inline docs/user_guides/debugging/stepping.rst:154]: Please, replace it with explanation from documentation.

**Review by PokhodenkoSA — CHANGES_REQUESTED** [review]
Please add `continue` at the end of commands scripts and check locally.

File: docs/user_guides/debugging/stepping.rst (hunk 7 above)
Comment (1e-to) [reply docs/user_guides/debugging/stepping.rst:154]: The documentation uses an example, so I only took the explanation from it.
Full info with example  is in see also link

**Comment by 1e-to** [comment]
run_debug_examples.sh tested locally, all works

**Review by PokhodenkoSA — APPROVED** [review]

**Pull request merged**